### DISCUSSION TITLE
Next ISR slow when there are hundred of thousands of generated pages

### DISCUSSION THREAD

**Discussion by lazidoca** [vercel/next.js#38674] [Help]
Hi,
I am using on-demand Incremental Static Regeneration to generate hundred of thousands product pages. I set `isrMemoryCacheSize = 0` to use  file system caching. When the number of pages increases, I notice the site becomes slower even pre-generated page. I checked the `.next` folder and saw the `.next/server/page/` contains thousand pages. 

Does large number of files cause the slow down? 

Old [discussion ](https://stackoverflow.com/questions/466521/how-many-files-can-i-put-in-a-directory) on SO

**Comment by officialrajdeepsingh** [comment]
Yes, it is true. When i convert my old ghost blog into nextjs, I face the same issue. I found content layer package is a perfect solution for my blog.

**Reply by lazidoca** [reply]
I think you are mentioning the slow build time when there are too many posts and it takes time to get the static props at build time. What I concerned was the slow of the generated files serving from a single directory

**Comment by opdrift** [comment]
I have been struggling with this issue too. If you are self-hosting your solution you could try to adjust a couple of things for your node environment.

I managed to improve the performance of next ISR returning files by using PM2 process manager to run the node process in cluster mode, thereby utilizing multiple cores in the CPU.

Also I increased the UV_THREADPOOL_SIZE to enable more threads to access files.

The combination of multiple cores and more threads made a big improvement on the access to the file system.
